### ISSUE TITLE
Support text-generation-webui via its openai API

### ISSUE THREAD

**Issue by Wehzie** [promptfoo/promptfoo#387]
The documentation doesn't list [text-gen-webui](https://github.com/oobabooga/text-generation-webui) as supported, however getting it to run isn't difficult as it has an OpenAI API [extension](https://github.com/oobabooga/text-generation-webui/wiki/12-%E2%80%90-OpenAI-API).

If it works, what is this issue about?
Currently, promptfoo doesn't forward additional `config` parameters to the API if they are outside the official OpenAI parameters.

The text-gen-webui extension can be activated from with the UI or via command line. Here is an example of command line usage.

```
python server.py --loader <LOADER-NAME> --model <MODEL-NAME> --api
#optionally replace `python server.py` with ./start_linux if simple installer is used
```

In promptfoo we can address the API as follows.

```
  - openai:chat:<MODEL-NAME>:
      id: <MODEL-ID>
      config:
        apiKey: foo
        apiBaseUrl: http://localhost:5000
        temperature: 0.8 # works
        max_tokens: 1024 # works
        mode: instruct # ignored
        instruction_template: LLama-v2 # ignored
```

We can test the API with curl, here the additional non-standard arguments are adhered to.

```
curl http://127.0.0.1:5000/v1/chat/completions \
  -H "Content-Type: application/json" \
  -d '{
    "messages": [
	  {
	    "role": "system",
	    "content": "You give pizza recipies."
	  },
      {
        "role": "user",
        "content": "Pepperoni pizza."
      }
    ],
    "mode": "instruct",
    "instruction_template": "Llama-v2",
    "temperature": 0.0,
    "max_tokens": 200
  }'
```
Is a pass through of arguments feasible?

**Comment by typpo** [comment]
Thanks for the suggestion, could definitely see this being useful. #388 adds this functionality, I will land it once I update the documentation too.

Right now the implementation would allow for the following:

```yaml
  - openai:chat:<MODEL-NAME>:
      id: <MODEL-ID>
      config:
        apiKey: foo
        apiBaseUrl: http://localhost:5000
        temperature: 0.8 
        max_tokens: 1024 
        passthrough:
          mode: instruct 
          instruction_template: LLama-v2 
          # anything else you want...
```
alternatively I could make it work without the `passthrough` key like in your original example.  Just a bit messier in the code..

**Comment by Wehzie** [comment]
Passthrough looks good. 
Thank you and happy new year!

**Comment by typpo** [comment]
`passthrough` support landed in 0.35.1, and I've updated the documentation
